pkg/deviceplugin/server.go

```diff
@@ -204,11 +217,17 @@ func (srv *server) setupAndServe(namespace string, devicePluginPath string, kube
 		if err = watchFile(pluginSocket); err != nil {
 			return err
 		}
-		fmt.Printf("Socket %s removed, restarting\n", pluginSocket)
+		action := "restarting"
+		if srv.state != serving {
```

Comment:
I'd prefer to avoid negations in conditions to lessen cognitive load. Better use 
```
if srv.state == serving {
...
}
```
This way we can also move
```
srv.grpcServer.Stop()
os.Remove(srv.pluginSocket)
``` 
into the conditional branch, because in the `terminating` state the gRPC server has been stopped already as well as the socket file has been removed.